### PR TITLE
Add email verification flow

### PR DESCRIPTION
**Pull request by maebeam**
I also cleaned up the large number of params we were passing to APIServer.

### CODE REVIEW

File: routes/verify.go [hunk 1]
```diff
@@ -318,3 +321,94 @@ func (fes *APIServer) SubmitPhoneNumberVerificationCode(ww http.ResponseWriter,
...
+	validHash := fes.verifyEmailHash(userMetadata.Email, requestData.PublicKey)
+	if requestData.EmailHash != validHash {
+		_AddBadRequestError(ww, fmt.Sprintf("VerifyEmail: Invalid hash: %s", requestData.EmailHash))
+		return
+	}
+
+	userMetadata.EmailVerified = true
+
+	err = fes.putUserMetadataInGlobalState(userMetadata)
+	if err != nil {
```
Comment (lazynina) [inline routes/verify.go:360]:
```suggestion
	if err := fes.putUserMetadataInGlobalState(userMetadata); err != nil {
```

File: routes/verify.go [hunk 2]
```diff
@@ -318,3 +321,94 @@ func (fes *APIServer) SubmitPhoneNumberVerificationCode(ww http.ResponseWriter,
...
+		_AddBadRequestError(ww, fmt.Sprintf("VerifyEmail: Failed to save user: %v", err))
+		return
+	}
```
Comment (lazynina) [inline routes/verify.go:363]: Do we need to send back some sort of 200 response at the end of this function? or does it automatically return a 200 if nothing is sent back?

**Review by lazynina — APPROVED** [review]
this looks good to me.

File: routes/verify.go (hunk 2 above)
Comment (maebeam) [reply routes/verify.go:363]: 200, empty response is the default

**Pull request merged**